support-files/mariadb.service

```diff
@@ -0,0 +1,104 @@
+#
+#  /etc/systemd/system/mariadb.service
+#
+
+[Unit]
+Description=MariaDB database server
+After=network.target
+After=syslog.target
+Requires=%p.socket
+
+[Install]
+WantedBy=multi-user.target
+Alias=mysql.service
+Alias=mysqld.service
+
+
+[Service]
+
+##############################################################################
+## Core requirements
+##
+
+Type=Notify
+
+# Setting this to true can break replication and the Type=Notify settings
+PrivateNetwork=false
+
+##############################################################################
+## Package maintainers
+##
+
+User=mysql
+
+# Execute pre and post scripts as root
+PermissionsStartOnly=true
+
+# Needed to create system tables etc.
+ExecStartPre=/usr/bin/mariadb-systemd-start
+
+# Start main service
+# skip-networking as its socket activation based
+ExecStart=/usr/sbin/mysqld $OPTIONS
+
+KillMode=process
+KillSignal=SIGTERM
+# Don't want to see an automated SIGKILL ever
+SendSIGKILL=no
+
+RestartPreventExitStatus=1
+Restart=on-fail
+RestartSec=5s
+
+PrivateTmp=true
+PrivateDevices=true
+
+UMask=077
+WorkingDirectory=/
+# To avoid mysql reading /root/.my.cnf which according to debian init script is
+# problematic
+Environment="HOME="
+
+##############################################################################
+## USERs can override
+##
+##
+## by creating a file in /etc/systemd/system/mariadb.server.d/MY_SPECIAL.conf
+
+# Useful options not previously available in [mysqld_safe]
+
+OOMScoreAdjust=-600
+
+BlockIOWeight=1000
+
+##
+## Options previously available to be set via [mysqld_safe]
+## that now needs to be set by systemd config files as mysqld_safe
+## isn't executed.
+##
```

Comment:
instead of many ExecStartPre's below I'd suggest one
ExecStartPre=mariadb-exec-start script that would read [mysqld_safe] section of
my.cnf do whatever is specified there.
ok --interleave cannot be done that way, but the rest will
work, I suppose?

when this is done what mysqld_safe functionality will be missing?
auto-restarts? what else?